salt/modules/dockermod.py

```diff
@@ -6728,9 +6727,23 @@ def call(name, function, *args, **kwargs):
         name, thin_path, os.path.join(thin_dest_path, os.path.basename(thin_path))
     )
 
+    # figure out available python interpreter inside the container (only Python3)
+    pycmds = ("python3", "/usr/libexec/platform-python")
+    container_python_bin = None
+    for py_cmd in pycmds:
+        cmd = [py_cmd] + ["--version"]
+        ret = run_all(name, subprocess.list2cmdline(cmd))
```

Comment:
apologies for not catching this earlier, but does this work if we just pass cmd as a list and pass in `python_shell=False`?

Reply:
Unfortunatetly it seems it does not work passing a list and `python_shell=False`:

```
2021-06-02 13:17:36,119 [salt.loaded.int.module.dockermod:6882][ERROR   ][31292] ret -> {'pid': 31702, 'retcode': 127, 'stdout': '', 'stderr': 'env: ‘[python3,’: No such file or directory'}
2021-06-02 13:17:37,495 [salt.loaded.int.module.dockermod:6882][ERROR   ][31292] ret -> {'pid': 31737, 'retcode': 127, 'stdout': '', 'stderr': 'env: ‘[/usr/libexec/platform-python,’: No such file or directory'}
```

Actually all the rest of `run_all` calls inside this module are performed also like this one, calling `subprocess.list2cmdline(cmd)` to convert the list to string.

The `dockermod` module uses the `run_all` function to execute commands inside the containers. This function eventually executes `container_resource.run` which is expecting a command string and not a list to inject into the `docker exec`.

Reply:
Thanks for confirming.